editor/editor.tsx

```diff
@@ -461,15 +480,15 @@ function Editor(props: any) {
                         </span>
                     </>
                 )}
-                <span className="demo-dropdown" hidden={urlSpec !== null || urlGist !== null}>
+                <span className="demo-dropdown" hidden={urlSpec !== null || urlGist !== null || urlExampleId !== null}>
                     <select
                         style={{ maxWidth: IS_SMALL_SCREEN ? window.innerWidth - 180 : 'none' }}
                         onChange={e => {
-                            setDemo(examples.find(d => d.id === e.target.value) as any);
+                            setDemo(examples[e.target.value] as any);
                         }}
-                        defaultValue={demo.id}
+                        value={demo.id}
```

Comment:
> About the example structure, well, I am okay to change it back. Just feel a dict is easier to access examples compared with an array.

I am okay with the new structure, and I think it has an advantage that it prevents us from accidentally assigning the same `id` to multiple examples. 

Since `id` and `key` are duplicated currently, how about removing `id` properties? I think `demo.id` in this line can be replaced with `urlExampleId` and we could change the following

```js
const SHOWN_EXAMPLE_LIST = Object.values(examples).filter(d => !d.hidden);
```

to

```js
const SHOWN_EXAMPLE_ENTRIES = Object.entries(examples).filter([k, d] => !d.hidden);
```

to keep the keys ([Object.entries()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)).

Reply:
Sounds great! Will change it accordingly